internal/templates/antiaffinity/template.go

```diff
@@ -0,0 +1,45 @@
+package antiaffinity
+
+import (
+	"fmt"
+
+	"golang.stackrox.io/kube-linter/internal/check"
+	"golang.stackrox.io/kube-linter/internal/diagnostic"
+	"golang.stackrox.io/kube-linter/internal/extract"
+	"golang.stackrox.io/kube-linter/internal/lintcontext"
+	"golang.stackrox.io/kube-linter/internal/objectkinds"
+	"golang.stackrox.io/kube-linter/internal/templates"
+	"golang.stackrox.io/kube-linter/internal/templates/antiaffinity/internal/params"
+)
+
+func init() {
+	templates.Register(check.Template{
+		HumanName:   "Anti affinity not specified",
+		Key:         "anti-affinity",
+		Description: "Flag objects with multiple replicas but anti affinity not specified in the pod template spec",
+		SupportedObjectKinds: check.ObjectKindsDesc{
+			ObjectKinds: []string{objectkinds.DeploymentLike},
+		},
+		Parameters:             params.ParamDescs,
+		ParseAndValidateParams: params.ParseAndValidate,
+		Instantiate: params.WrapInstantiateFunc(func(_ params.Params) (check.Func, error) {
+			return func(_ *lintcontext.LintContext, object lintcontext.Object) []diagnostic.Diagnostic {
+				replicas, found := extract.Replicas(object.K8sObject)
+				if !found {
+					return nil
+				}
+				if *replicas > 1 {
+					podTemplateSpec, hasPods := extract.PodTemplateSpec(object.K8sObject)
+					if !hasPods {
+						return nil
+					}
+					if podTemplateSpec.Spec.Affinity != nil && podTemplateSpec.Spec.Affinity.PodAntiAffinity != nil {
+						return nil
+					}
```

Comment:
So what I had in mind for this check was something even more strict: specify that intra-pod anti-affinity is set for pods of this deployment, to ensure that they would be scheduled on different replicas.

Mechanically, this means ensuring that at least one of the terms in the `podAntiAffinity` (whether it's `required` or `prefered`) satisfies _both_ of the following properties:
1. topologyKey is `kubernetes.io/hostname` (so that anti-affinity is based on nodes)
2. the labelSelector matches the labels in the pod template spec (that is, it would match pods generated from the current spec)

See the following for an example (copied from [here](https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#always-co-located-in-the-same-node)) of a deployment that _would_ pass this check.
```
apiVersion: apps/v1
kind: Deployment
metadata:
  name: redis-cache
spec:
  selector:
    matchLabels:
      app: store
  replicas: 3
  template:
    metadata:
      labels:
        app: store
    spec:
      affinity:
        podAntiAffinity:
          requiredDuringSchedulingIgnoredDuringExecution:
          - labelSelector:
              matchExpressions:
              - key: app
                operator: In
                values:
                - store
            topologyKey: "kubernetes.io/hostname"
      containers:
      - name: redis-server
        image: redis:3.2-alpine

```